PrioritiesSprint3.md

```diff
@@ -0,0 +1,66 @@
+These are the priorities for Sprint number 3, which runs from April 29, 2024 to May 26, 2024. 
+In this sprint, the Eel build is in CI and the Octopus build is in Test. Each priority has a list of subtasks that are to be completed within this sprint.
+
+Each priority has a list of subtasks that are to be completed within this sprint.  
+
+Note that ARAs will be required to implement either MCQ or Pathfinder (or both).
+
+# Develop Pathfinder Query
+## Pre-development work:
+- Design new landing page to distinguish types of interactions (templated query, pathfinder, multiquery(?)) (UI)
+- Design query selection interface - Dependencies: decisions on categories of nodes to be allowed as starting points and any other factors allowed for query selection (UI, ARAs)
+- User testing path grouping strategies (UI, ARAs)
+- Design results page path  and graph display (similar to results now) - Dependencies: understanding of the number of paths and how they will be grouped (UI, ARAs)
+- Research to identify new facets needed. e.g. facets for other categories - Dependencies: pathfinder data examples (UI, ARAs)
+- User research filter designs for new filters like metaedge type or other. (UI)
+- Provides example datasets to UI (ARAs)
+## Octopus (CI) work:
+- Evaluate performance and result quality from prototype implementations (ARAs)
+
+# MultiCURIE Query (MCQ)
+## Pre-development work:
+- Designs multi curie selection -  requires interface to add and subtract select curies (how many max?) (UI)
+- Investigate additional facets needed ( e.g. curie match facet, other specific to starting or ending node category if neccesary) (UI)
+- User research filter designs for new filters like metaedge type or other. (UI)
+- ## Eel (CI):
+- Relevant KPs implement lookup MCQ (KPs)
+- NodeNorm implements set ID method
+## Octopus (TEST):
+- TRAPI 1.5 finalized and released (ARAs, KPs, UI)
+
+# Display Full KL/AT in the UI
+## Pre-Development
+- ARAs/O&O plans for utilizing KL/AT information in their scoring approaches (ARA, O&O)
+- Define next EPC area to standardize and implement in data and UI. (EPC)
+- ## Eel (CI):
+- Last few KPs/ARAs annotate edges with KL/AT terms (See status table [here](https://docs.google.com/document/d/16agzJB0OlR8z-zU3nTivYYIo09sdayN-tu8lbrZyTXY/edit?pli=1#bookmark=id.kz0zera1i398)) (KPs, ARAs)
+## Octopus (TEST):
+- UI team implements full KL/AT tags in UI display (define paradigm to accommodate this info, which may include user friendly synonyms, new icons/badges, updated EPC modal tables, etc). (UI)
+- Determine how to display KL/AT for KPs/ARAs that have not yet implemented edge-level annotations (may fall back to infores-based annotation?) (UI)
+
+# Treats Refactor
+## Octopus (TEST):
+- UI implements "treats" refactor including provenance (UI)
+
+# Scalability
+## Eel (CI):
+- Sending Concurrent queries (MVP1&MVP2) in batch of 5,10,15,20,25,30 (Test Env)
+
+# CQS Workflows
+## Eel (CI):
+-  Test and iterate MVP1 workflows
```

Comment:
Here are the details of "how to test MVP1 workflows with the CQS":
- run every CQS template through DEV WFR and confirm that no errors are produced.  Could be that there are no records returned, and that is ok. 
- make sure that each CQS template passes TRAPI JSON validation - maybe we can incorporate reasoner-validator tests into the CQS template repo, but if not, at least manually.
------- MERGE TEMPLATES INTO CI -------------
- Solicit at least 3 Q/A pairs from each group that has a CQS template
- Create a series of TestCases (and a TestSuite)
- Modify the TestRunner to run the CQS TestCases (and TestSuite) against the CI version of CQS (or at least against CI ARS?)?
- Assemble a list of PKs to examine manually from CI
- Examine each PK to see that SPO + support edges are filled out with the provenance we expect in CI
------- MERGE TEMPLATES INTO TEST----------

Reply:
Here's the protocol for submitting new CQS templates: https://github.com/TranslatorSRI/CQS/tree/main/templates. I don't think we want this URL or the details above in the PR.